src/validation.rs

```diff
@@ -69,6 +78,24 @@ pub fn is_rfc_1123_subdomain(value: &str) -> Vec<String> {
     errors
 }
 
+/// Tests for a string that conforms to the definition of a label in DNS (RFC 1035).
+pub fn is_rfc_1035_label(value: &str) -> Vec<String> {
```

Comment:
(For subdomain as well,) I do not like the "is_xyz" not returning a bool. 
For the usage: If the return value is empty, it is a valid label. I find that a bit strange.
I see that you wanna collect and propagate the errors.

What about a tuple:
pub fn is_rfc_1035_label(value: &str) -> (bool, Option<Vec<String>>)
Result with custom error containing a vec?
pub fn is_rfc_1035_label(value: &str) -> Result<bool, RfcError)

Reply:
What about `Result<(), Vec<String>>`?

The best would be to avoid boolean blindness and parse instead of validate the input, e.g.
```
struct Label(String);

type ParseErrors = Vec<String>;

pub fn parse_rfc_1035_label(value: &str) -> Result<Label, ParseErrors> {
    …
}
```

Reply:
I'm mostly with Sigi here, Result<(),Vec<String>> would have been my first choice. 
Parsing seems a bit overkill to me, as we basically return a String if I am not mistaken? Defining an extra struct for that seems like overkill to me.

Reply:
I agree that this isn't a great API.
I copied it almost verbatim from Kubernetes: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L187

I agree that it'd be great - in theory - to have all usages automatically validated by using those special types but I'm not sure how realistic that is considering that our input and output is always "untyped".....but I like it.

I think for now the best would be to either keep it as it is for maximum "compatibility" with K8S or use the `Result<()...>` thing.

Reply:
Do we gain anything by sticking to the way Kubernetes does it?
Unless there is a clear benefit I'd be in favor of doing this the rusty way and using a Result